### PR TITLE
Wrote a report on the different CV algorithms that could replace our current method

### LINKED ISSUE #66: Research Simplified CV Algorithms for Efficient Sheet Processing
Description: Our current method for processing Scantron sheets, while providing 100% accuracy, is relatively complex and may become cumbersome as we expand the application to support custom sheet formats. To ensure scalability and maintain the application's performance, we need to research and explore more efficient computer vision (CV) algorithms that simplify the process without compromising accuracy.

Goal:
Identify CV algorithms that offer a more lightweight and streamlined approach for processing sheets, particularly as we introduce support for custom-designed sheets.
Ensure the selected approach maintains or improves current accuracy levels while optimizing performance to reduce computational load.

Tasks:
Research and evaluate existing CV algorithms known for efficiency in form processing.
Compare them against our current method, focusing on accuracy, processing speed, and resource usage and implementation complexity.

Success Criteria:
A more efficient CV process is identified that simplifies the current method without sacrificing accuracy.

### PR DESCRIPTION
**Pull request by ramezmosad**
Fixes #66 

**What was changed?**
*Added a readme file, named "algorithms.md" in the inputData folder in the BubbleScan-AI folder.*

**Why was it changed?**
*I did not write any code, but the report I wrote compared three different methods against our current method for efficiency in form processing.*

**How was it changed?**
*I did not modify any files, I just created a new markdown file in the BubbleScan-AI/inputData/ folder.*

### CODE REVIEW

File: inputData/algorithms.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+# Evaluating Efficient CV Algorithms for Form Processing
+
+
```
Comment (yildizahmetali) [inline inputData/algorithms.md:3]: Nicely organized

File: inputData/algorithms.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+While CNNs offer potential benefits in robustness and adaptability, the increased complexity and resource requirements make it a secondary consideration.
+
+**Conditions for Adoption:**
+
+- **Availability of Data:** If we can generate a sufficiently large and diverse dataset through synthetic generation and augmentation.
+- **Resource Allocation:** If computational resources (e.g., GPUs) and expertise in machine learning are available.
+- **Long-Term Scalability:** If we anticipate frequent changes in form designs and need a highly adaptable system.
+
+---
+
```
Comment (yildizahmetali) [inline inputData/algorithms.md:223]: It is nice that you added the references

**Review by yildizahmetali — DISMISSED** [review]
Nicely Done!

**Comment by kate-holdener** [comment]
This is a very nice report, the summary table is super informative. I'd vote for trying the "Edge Detection & Perspective Transformation" due to high accuracy and fast processing. I don't think that image quality will be a problem.

**Review by Chirag2x — COMMENTED** [review]
Good Work @ramezmosad. The document is well explained, Initially I also wanted to opt for CNN based method but Edge Detection would be a great choice due to being lightweight and well optimised structure. We will discuss this further in our meeting. Keep it up!

**Review by Chirag2x — APPROVED** [review]
Good work!!

**Pull request merged**